### PR TITLE
Add option to commit and push changes resulting from running a command.

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: for-all-boosters.sh [hunk 1]
```diff
@@ -474,6 +474,29 @@ catalog() {
...
 }
 
+run_cmd() {
+    # build command: since arguments are passed unquoted, we need a separator to mark the command from the commit message
+    # we use ---- as separator so we expect arguments to be in the form "cmd ---- msg"
+    # first build a string from all arguments
+    local cmdAndMsg=$(echo "$@")
+    # then we extract parts using https://stackoverflow.com/a/10520718
+    local cmd=${cmdAndMsg% ----*} # note the space after % to avoid extracting the space before the separator into cmd
+    local msg=${cmdAndMsg#*---- } # note the space after the separator to avoid extracting it into msg
```
Comment (metacosm) [inline for-all-boosters.sh]: Doesn't work if there is no msg…

Comment (geoand) [reply for-all-boosters.sh]: So there needs to be some kind of validation before this is run?

File: for-all-boosters.sh [hunk 2]
```diff
@@ -669,9 +699,29 @@ case "$subcommand" in
         cmd="run_smoke_tests"
     ;;
     cmd)
-        shift
+        # Needed in order to "reset" the options processing for the subcommand
+        OPTIND=2
```
Comment (geoand) [inline for-all-boosters.sh]: Does this work correctly with other flags we have? I mean it doesn't interfere, does it?

Comment (metacosm) [reply for-all-boosters.sh]: I think it does but I need to test it more. We really need some sort of unit testing for our scripts :)

File: for-all-boosters.sh (hunk 1 above)
Comment (metacosm) [reply for-all-boosters.sh]: Should be fixed now.

File: for-all-boosters.sh (hunk 2 above)
Comment (geoand) [reply for-all-boosters.sh]: Unit testing for scripts? Does such a thing exist? :P

**Pull request merged**